VMAccess/README.md

```diff
@@ -0,0 +1,252 @@
+```powershell
+Switch-AzureMode -Name AzureResourceManager
+```
+
+You can deploying VMAccess Extension by running:
+```powershell
+$RGName = '<resource-group-name>'
+$VmName = '<vm-name>'
+$Location = '<location>'
+
+$ExtensionName = 'VMAccessForLinux'
+$Publisher = 'Microsoft.OSTCExtensions'
+$Version = <version>
+
+$PublicConf = '{}'
+$PrivateConf = '{
+  "username": "<username>",
+  "password": "<password>",
+  "ssh_key": "<pem-cert-contents>",
+  "reset_ssh": true|false,
+  "remove_user": "<username-to-remove>"
+}'
+
+Set-AzureVMExtension -ResourceGroupName $RGName -VMName $VmName -Name $ExtensionName -Publisher $Publisher -ExtensionType $ExtensionName -TypeHandlerVersion $Version -Settingstring $PublicConf -$PublicConf $PrivateConf -Location $Location
+```
+
+For more details about Set-AzureVMExtension syntax in ARM mode, please visit [Set-AzureVMExtension][Set-AzureVMExtension-ARM].
+
+### 2.3. Using [**ARM Template**][arm-template]
+```json
+{
+  "type": "Microsoft.Compute/virtualMachines/extensions",
+  "name": "<extension-deployment-name>",
+  "apiVersion": "<api-version>",
+  "location": "<location>",
+  "dependsOn": [
+    "[concat('Microsoft.Compute/virtualMachines/', <vm-name>)]"
+  ],
+  "properties": {
+    "publisher": "Microsoft.OSTCExtensions",
+    "type": "VMAccessForLinux",
+    "typeHandlerVersion": "1.1",
+    "settings": {},
+    "protectedSettings": {
+      "username": "<username>",
+      "password": "<password>",
+      "reset_ssh": true,
+      "ssh_key": "<ssh-key>",
+      "remove_user": "<username-to-remove>"
+    }
+  }
+
+}
+```
+
+For more details about ARM template, please visit [Authoring Azure Resource Manager templates](https://azure.microsoft.com/en-us/documentation/articles/resource-group-authoring-templates/).
+
+## 3. Scenarios
+
+### 3.1 Resetting the password
+```json
+{
+  "username":"currentusername",
+  "password":"newpassword"
+}
+```
+
+### 3.2 Resetting the SSH key
+```json
+{ 
+  "username":"currentusername", 
+  "ssh_key":"contentofsshkey",   
+}
+```
+
+### 3.3 Resetting the password and the SSH key
+```json
+{
+  "username":"currentusername",
+  "ssh_key":"contentofsshkey",
+  "password":"newpassword",
+}
+```
+
+### 3.4 Creating a new sudo user account
+```json
+{
+  "username":"newusername",
+  "password":"newpassword"
+}
+```
+
+### 3.5 Resetting the SSH configuration
+```json
+{
+  "reset_ssh": true
+}
+```
+
+### 3.6 Removing an existing user
+```json
+{
+  "remove_user":"usertoberemoveed",
+}
+```
+
+
+## Supported Linux Distributions
+- Ubuntu 12.04 and higher
+- CentOS 6.5 and higher
+- Oracle Linux 6.4.0.0.0 and higher
+- openSUSE 13.1 and higher
+- SUSE Linux Enterprise Server 11 SP3 and higher
+- FreeBSD
+
+## Debug
+
+* The status of the extension is reported back to Azure so that user can
+see the status on Azure Portal
+* The operation log of the extension is `/var/log/azure/<extension-name>/<version>/extension.log` file.
+
+
```

Comment:
Do you mean the links of azure-powershell, azure-cli?
These lines should not be seen because it is just a alias for the link. And the alias is referenced in the document.